Add new Insert Before/After menu items and shortcuts

talldan opened the pull request:
## Description
Closes #7297

Adds the following keyboard shortcuts for those actions:
- <kbd>Option</kbd>+<kbd>Cmd</kbd>+<kbd>T</kbd> (Mac) / <kbd>Ctrl</kbd>+<kbd>Alt</kbd>+<kbd>T</kbd> - Insert Before
- <kbd>Option</kbd>+<kbd>Cmd</kbd>+<kbd>Y</kbd> (Mac) / <kbd>Ctrl</kbd>+<kbd>Alt</kbd>+<kbd>Y</kbd> - Insert After

## How has this been tested?
- Tested use of MenuItem and shortcut across browsers on Mac (Chrome, Safari, Firefox, Opera)
- Tested use of MenuItem and shortcut across browsers on Windows (Chrome, IE11, Edge, Firefox, Opera)
- Tested with a block that uses InnerBlocks with a template lock

## Screenshots <!-- if applicable -->
**MenuItems**
![screen shot 2018-08-13 at 4 33 36 pm](https://user-images.githubusercontent.com/677833/44021434-683edc54-9f17-11e8-9c91-fe0f8c1e0eb1.png)

**Shortcut Help**
![screen shot 2018-08-13 at 4 33 01 pm](https://user-images.githubusercontent.com/677833/44021471-7a31a806-9f17-11e8-81ec-38213dfbeefb.png)

## Types of changes
<!-- What types of changes does your code introduce?  -->
<!-- Bug fix (non-breaking change which fixes an issue) -->
New feature (non-breaking change which adds functionality)
<!-- Breaking change (fix or feature that would cause existing functionality to not work as expected) -->

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->

talldan committed:
Add basic insert before and insert after buttons to block settings menu

talldan committed:
Add keyboard shortcuts for insert before and insert after

talldan committed:
Change insert before/after shortcuts to meta+alt+t and meta+alt+y

talldan committed:
Add comment explaining use of preventDefault

talldan committed:
Add insert before/after shortcuts to keyboard help

jasmussen reviewed (approved):
Screenshot:

<img width="875" alt="screen shot 2018-08-13 at 11 24 33" src="https://user-images.githubusercontent.com/1204802/44023728-80513714-9eeb-11e8-8bd9-e2c79417569b.png">

Love it. Works as intended.

Might want a code sanity check by @noisysocks, but otherwise, ship it!

talldan commented:
Great - I still need to do a bit of work on the template locking side of things. I'll do that first and then get a code review

talldan committed:
Do not show Insert Before/After menu items when a block is template locked

talldan committed:
Disallow block settings actions that can be triggered through shortcuts and should not be possible when template locking is active

talldan committed:
Tidy up comments

- Add full-stops
- Remove comment that no longer seems relevant (component is not a DropDown)

talldan commented:
Ready for review now :)

noisysocks commented on packages/editor/src/components/block-settings-menu/index.js:
```diff
@@ -199,15 +235,32 @@ export default compose( [
+			lastSelectedIndex,
+			isLocked,
+			canDuplicate,
+		} = props;
```
Nit: you could put newlines into these function arguments if you don't want `props` in the scope.

```js
withDispatch( ( dispatch, {
	clientIds,
	rootClientId,
	// etc.
} ) ) {
	// etc.
}
```

noisysocks commented on packages/editor/src/components/block-settings-menu/index.js:
```diff
@@ -229,7 +282,19 @@ export default compose( [
-				removeBlocks( clientIds );
+				if ( ! isLocked ) {
+					removeBlocks( clientIds );
+				}
```
Nice, a sneaky bug fix!

noisysocks reviewed (approved):
Code looks fab, and everything works great in my testing!

I tested template locking using this custom post type:

```php
register_post_type( 'test_type', array(
	'label' => 'Test',
	'public' => true,
	'show_in_rest' => true,
	'template_lock' => 'insert',
	'template' => array(
		array( 'core/image' ),
		array( 'core/quote' ),
		array( 'core/image' ),
	),
) );
```

And this custom block:

```js
registerBlockType( 'test/block', {
	title: 'Test block',
	category: 'common',
	save: () => null,
	edit() {
		return (
			<wp.editor.InnerBlocks
				templateLock="insert"
				template={ [ [ 'core/image' ], [ 'core/image' ] ] }
			/>
		);
	},
} );
```

Great job 👍👍👌:shipit:

mtias commented:
Nice one!
